File: ruby/hamming/hamming.rb
require 'pry'
class Hamming
  def self.compute(strand_a, strand_b)
    raise ArgumentError, "Strands must be indentical in length" unless strand_a.length == strand_b.length
    mismatches = 0
    strand_a.chars.each_with_index do |letter, index|
      mismatches += 1 if letter != strand_b[index]
    end
    mismatches
  end
end

File: javascript/hamming/hamming.js
var Hamming = function() {};

Hamming.prototype.compute = function(strandOne, strandTwo) {
  if (strandOne.length != strandTwo.length) { throw "DNA strands must be of equal length."};
  var differences = 0;
  for (i = 0; i < strandOne.length; i ++) {
    if (strandOne[i] !== strandTwo[i]) { differences += 1 };
  }
  return differences;
};

module.exports = Hamming;
